Tools/TestWebKitAPI/Tests/WebKitCocoa/LoadAndDecodeImage.mm

```diff
@@ -156,4 +159,46 @@
     Util::run(&done);
 }
 
+TEST(WebKit, GetInformationFromImageData)
+{
+    RetainPtr webView = adoptNS([WKWebView new]);
+    done = false;
+    RetainPtr pngData = [NSData dataWithContentsOfURL:[NSBundle.test_resourcesBundle URLForResource:@"icon" withExtension:@"png"]];
+    [webView _getInformationFromImageData:pngData.get() completionHandler:^(NSString *typeIdentifier, NSArray<NSValue *> *availableSizes, NSError *error) {
+        EXPECT_NULL(error);
+        EXPECT_TRUE([typeIdentifier isEqualToString:UTTypePNG.identifier]);
+        EXPECT_EQ(1u, availableSizes.count);
+        for (NSValue *size in availableSizes) {
+            EXPECT_EQ(215, [size sizeValue].width);
+            EXPECT_EQ(174, [size sizeValue].height);
+        }
+        done = true;
+    }];
+    Util::run(&done);
+
+    done = false;
+    RetainPtr gifData = [NSData dataWithContentsOfURL:[NSBundle.test_resourcesBundle URLForResource:@"apple" withExtension:@"gif"]];
+    [webView _getInformationFromImageData:gifData.get() completionHandler:^(NSString *typeIdentifier, NSArray<NSValue *> *availableSizes, NSError *error) {
+        EXPECT_NULL(error);
+        EXPECT_TRUE([typeIdentifier isEqualToString:UTTypeGIF.identifier]);
+        EXPECT_EQ(1u, availableSizes.count);
+        for (NSValue *size in availableSizes) {
+            EXPECT_EQ(52, [size sizeValue].width);
+            EXPECT_EQ(64, [size sizeValue].height);
+        }
+        done = true;
+    }];
+    Util::run(&done);
+
+    done = false;
+    RetainPtr svgData = [NSData dataWithContentsOfURL:[NSBundle.test_resourcesBundle URLForResource:@"AllAhem" withExtension:@"svg"]];
+    [webView _getInformationFromImageData:svgData.get() completionHandler:^(NSString *typeIdentifier, NSArray<NSValue *> *availableSizes, NSError *error) {
+        EXPECT_NULL(error);
+        EXPECT_TRUE([typeIdentifier isEqualToString:UTTypeSVG.identifier]);
+        EXPECT_EQ(0u, availableSizes.count);
+        done = true;
+    }];
+    Util::run(&done);
+}
```

Comment:
Should we add a test case for unsupported image types like JPEG2000 for example?

Reply:
Yes, added